### PR TITLE
Unuse deprecated

### PR DESCRIPTION
**Pull request by krux02**
``Utf16Char`` should really be ``uint16``, not ``int16``. Then the casts can disappear.

I tried to remove ``ze64`` from ccgexpr, but it is currently still the "cleanest solution". The proper way to fix it would be to convert ``TBitSet`` from ``seq[int8]`` to ``seq[byte]`` ([like the comment actually says](https://github.com/nim-lang/Nim/blob/aed8766e84364360e0ed1e1e667ee735afd99e81/compiler/bitsets.nim#L14)). But that would change too much for now.

I added ```proc ord*(arg: Utf16Char): int = int(cast[uint16](arg))```, because that is used for ``char`` as well. For consistency it only makes sense to provide it. If you don't want it, I can also unexport it and leave it private.

I also normalized the declaration of ``countSetBits``. It is now declared only once.

### CODE REVIEW

File: lib/system/widestrs.nim [hunk 1]
```diff
@@ -17,6 +17,8 @@ type
   Utf16Char* = distinct int16
   WideCString* = ref UncheckedArray[Utf16Char]
 
+proc ord*(arg: Utf16Char): int = int(cast[uint16](arg))
```
Comment (Araq) [inline lib/system/widestrs.nim]: Meh, don't export it, bugfixes shouldn't add new procs.

Comment (krux02) [reply lib/system/widestrs.nim]: yes I can unexport it again, would not be a problem. The question is, should it be exported? If you ask me, then yes it should.

Comment (Araq) [reply lib/system/widestrs.nim]: It shouldn't. ``ord`` is what it is and it's not some user-extensible conversion proc. It's bad enough that it exists at all, it serves little purpose over a type conversion to ``int``.

**Comment by Araq** [comment]
Please rebase so that we can see more easily if all tests are green.

**Comment by krux02** [comment]
bootstrapping is failing. This is the line in the generated sourcecode that is different:

```c
// correct
if (!(((16777280 &(1U<<((packSym->kind) & 31U)))!=0))) raiseFieldError(((NimStringDesc*) &TM_X9bGGOoNyKehOTF7f2xys3g_17));
// now
if (!(((     576 &(1U<<((packSym->kind) & 31U)))!=0))) raiseFieldError(((NimStringDesc*) &TM_X9bGGOoNyKehOTF7f2xys3g_17));
```

Where ``16777280`` is ``1<<26 | 1 << 6`` and ``576`` is ``1<<9 | 1<<6``. That is one bit set incorrectly.

File: lib/pure/bitops.nim [hunk 2]
```diff
@@ -76,26 +73,32 @@ when defined(nimHasalignOf):
...
+  template setBits*(v: typed): untyped =
+    ## Returns ``v``, with the bits at positions ``bits`` set to 1
+    discard
+
+  template setBits*(v: typed, bit: typed, bits: varargs[typed]): untyped =
     ## Returns ``v``, with the bits at positions ``bits`` set to 1
-    bits.expectKind(nnkBracket)
-    result = newStmtList()
-    for bit in bits:
-      result.add newCall("setBit", v, bit)
```
Comment (Araq) [inline lib/pure/bitops.nim]: This for loop is easier to understand than your C++ inspired varargs shift recursion with the ``setBits`` taking a single argument.

**Comment by Araq** [comment]
Sorry but this doesn't improve anything, it just shifts problems around. The little code duplication is preferable over the ``import bitops`` hack to me.

**Pull request closed without merging**